### PR TITLE
Initial version Grpc.Tools integration tests

### PR DESCRIPTION
**Pull request by jtattermusch**
#31638 with some fixes/cleanup/simplification on top of it.

### CODE REVIEW

File: src/csharp/Grpc.Tools.Tests/IntegrationTests/Directory.Build.targets [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  since we want different settings for the integration tests, particularly
+  we don't want to sign assemblies
+-->
+<Project>
+
+  <!-- These would normally be imported from the Grpc.Tools NuGet package -->
+  <Import Project="$(GrpcToolsBuildDir)\_grpc\_Grpc.Tools.targets"/>
+  <Import Project="$(GrpcToolsBuildDir)\_protobuf\Google.Protobuf.Tools.targets"/>
+
+  <!-- Needed for the net45 build to work on Unix. See https://github.com/dotnet/designs/pull/33 -->
```
[the remark this answers is not in the document]
Comment (jtattermusch) [reply src/csharp/Grpc.Tools.Tests/IntegrationTests/Directory.Build.targets]: Looks like it's safe to remove.

File: src/csharp/Grpc.Tools.Tests/Grpc.Tools.Tests.csproj [hunk 2]
```diff
@@ -7,12 +7,21 @@
...
   <Import Project="..\Grpc.Tools\SourceLink.csproj.include" />
 
+  <ItemGroup>
+    <!-- Prevent tests data files from being compiled into this project -->
+    <!-- TODO(jtattermusch): find a better solution for this. -->
+    <Compile Remove="IntegrationTests\**" />
+    <EmbeddedResource Remove="IntegrationTests\**" />
+    <None Remove="IntegrationTests\**" />
+  </ItemGroup>
+
```
Comment (tonydnewell) [inline src/csharp/Grpc.Tools.Tests/Grpc.Tools.Tests.csproj:17]: I did have the tests in a separate directory outside of the project but moved them back under the project to make it self contained. We could move them again if needed then these lines can be removed.

**Comment by tonydnewell** [comment]
Looks good.

**Comment by jtattermusch** [comment]
Closing in favor of https://github.com/grpc/grpc/pull/31638 (I pushed the same changes there as well).

**Pull request closed without merging**